pincer/commands/commands.py

```diff
@@ -705,6 +739,133 @@ async def add_commands(self, commands: List[AppCommand]):
         """
         await gather(*map(lambda cmd: self.add_command(cmd), commands))
 
+    def __build_local_commands(self):
+        """Builds the commands into the format that Discord expects. See class info
+        for the reasoning.
+        """
+        for cmd in ChatCommandHandler.register.values():
+
+            if cmd.sub_group:
+                # If a command has a sub_group, it must be nested to levels deep.
+                #
+                # command
+                #     subcommand-group
+                #         subcommand
+                #
+                # The children of the subcommand-group object are being set to include
+                # `cmd` If that subcommand-group object does not exist, it will be
+                # created here. The same goes for the top-level command.
+                #
+                # First make sure the command exists. This command will hold the
+                # subcommand-group for `cmd`.
+
+                # `key` represents the hash value for the top-level command that will
+                # hold the subcommand.
+                key = _hash_app_command_params(
+                    cmd.group.name,
+                    cmd.app.guild_id,
+                    AppCommandType.CHAT_INPUT,
+                    None,
+                    None,
+                )
+
+                if key not in ChatCommandHandler.built_register:
+                    ChatCommandHandler.built_register[key] = AppCommand(
+                        name=cmd.group.name,
+                        description=cmd.group.description,
+                        type=AppCommandType.CHAT_INPUT,
+                        guild_id=cmd.app.guild_id,
+                        options=[]
+                    )
+
+                # The top-level command now exists. A subcommand group now if placed
+                # inside the top-level command. This subcommand group will hold `cmd`.
+
+                children = ChatCommandHandler.built_register[key].options
+
+                sub_command_group = AppCommandOption(
+                    name=cmd.sub_group.name,
+                    description=cmd.sub_group.description,
+                    type=AppCommandOptionType.SUB_COMMAND_GROUP,
+                    options=[]
+                )
+
+                # This for-else makes sure that sub_command_group will hold a reference
+                # to the subcommand group that we want to modify to hold `cmd`
+
+                for cmd_in_children in children:
+                    if (
+                        cmd_in_children.name == sub_command_group.name
+                        and cmd_in_children.description == sub_command_group.description
+                        and cmd_in_children.type == sub_command_group.type
```

Comment:
isn't possible to just have check `cmd_in_children == sub_command`?

Reply:
no. Equality is super weird here since things that would be considered not equal would actually be the same command in Discord.